genesyscloud/resource_genesyscloud_architect_emergencygroup.go

```diff
@@ -0,0 +1,308 @@
+	sdkConfig := meta.(*providerMeta).ClientConfig
+	architectApi := platformclientv2.NewArchitectApiWithConfig(sdkConfig)
+
+	log.Printf("Reading emergency group %s", d.Id())
+	return withRetriesForRead(ctx, 30*time.Second, d, func() *resource.RetryError {
+		emergencyGroup, resp, getErr := architectApi.GetArchitectEmergencygroup(d.Id())
+		if getErr != nil {
+			if isStatus404(resp) {
+				return resource.RetryableError(fmt.Errorf("Failed to read emergency group %s: %s", d.Id(), getErr))
+			}
+			return resource.NonRetryableError(fmt.Errorf("Failed to read emergency group %s: %s", d.Id(), getErr))
+		}
+
+		if emergencyGroup.State != nil && *emergencyGroup.State == "deleted" {
+			d.SetId("")
+			return nil
+		}
+
+		d.Set("name", *emergencyGroup.Name)
+		d.Set("division_id", *emergencyGroup.Division.Id)
+
+		if emergencyGroup.Description != nil {
+			d.Set("description", *emergencyGroup.Description)
+		} else {
+			d.Set("description", nil)
+		}
+
+		if emergencyGroup.Enabled != nil {
+			d.Set("enabled", *emergencyGroup.Enabled)
+		} else {
+			d.Set("enabled", nil)
+		}
+
+		if emergencyGroup.EmergencyCallFlows != nil && len(*emergencyGroup.EmergencyCallFlows) > 0 {
+			d.Set("emergency_call_flows", flattenEmergencyCallFlows(*emergencyGroup.EmergencyCallFlows))
+		} else {
+			d.Set("emergency_call_flows", nil)
+		}
+
+		log.Printf("Read emergency group %s %s", d.Id(), *emergencyGroup.Name)
+		return nil
+	})
+}
+
+func updateEmergencyGroup(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	name := d.Get("name").(string)
+	description := d.Get("description").(string)
+	divisionId := d.Get("division_id").(string)
+	enabled := d.Get("enabled").(bool)
+
+	sdkConfig := meta.(*providerMeta).ClientConfig
+	architectAPI := platformclientv2.NewArchitectApiWithConfig(sdkConfig)
+
+	diagErr := retryWhen(isVersionMismatch, func() (*platformclientv2.APIResponse, diag.Diagnostics) {
+		// Get current emergency group version
+		emergencyGroup, resp, getErr := architectAPI.GetArchitectEmergencygroup(d.Id())
+		if getErr != nil {
+			return resp, diag.Errorf("Failed to read emergency group %s: %s", d.Id(), getErr)
+		}
+
+		log.Printf("Updating emergency group %s", name)
+		_, resp, putErr := architectAPI.PutArchitectEmergencygroup(d.Id(), platformclientv2.Emergencygroup{
+			Name:               &name,
+			Division:           &platformclientv2.Division{Id: &divisionId},
+			Description:        &description,
+			Version:            emergencyGroup.Version,
+			State:              emergencyGroup.State,
+			Enabled:            &enabled,
+			EmergencyCallFlows: buildSdkEmergencyGroupCallFlows(d),
+		})
+		if putErr != nil {
+			return resp, diag.Errorf("Failed to put emergency group %s: %s", d.Id(), putErr)
+		}
+		return resp, nil
+	})
+
+	if diagErr != nil {
+		return diagErr
+	}
+
+	log.Printf("Finished updating emergency group %s", name)
+	time.Sleep(5 * time.Second)
```

Comment:
And you can remove this sleep. The consistency checker will take care of eventual consistency issues